code/modules/shuttle/shuttle_manipulator.dm

```diff
@@ -0,0 +1,289 @@
+		L["timeleft"] = M.getTimerStr()
+		L["mode"] = capitalize(M.mode)
+		L["status"] = M.getStatusText()
+		if(M == existing_shuttle)
+			shuttle_data["existing_shuttle"] = L
+
+		shuttle_data["shuttles"] += list(L)
+
+	var/dat = ""
+	dat += "<center>"
+	dat += "<a href='?src=\ref[src];[src]=tab;tab=[SHUTTLE_MANIPULATOR_STATUS]' [current_tab == SHUTTLE_MANIPULATOR_STATUS ? "class='linkOn'" : ""]>Status</a>"
+	dat += "<a href='?src=\ref[src];[src]=tab;tab=[SHUTTLE_MANIPULATOR_TEMPLATE]' [current_tab == SHUTTLE_MANIPULATOR_TEMPLATE ? "class='linkOn'" : ""]>Templates</a>"
+	dat += "<a href='?src=\ref[src];[src]=tab;tab=[SHUTTLE_MANIPULATOR_MOD]' [current_tab == SHUTTLE_MANIPULATOR_MOD ? "class='linkOn'" : ""]>Modification</a>"
+	dat += "</center>"
+	dat += "<HR>"
+
+	if(busy)
+		dat += "<div class='statusDisplay'>Shuttle Manipulation in progress...</div>"
+	else
+		dat += "<div class='statusDisplay'>"
+
+		switch(current_tab)
+			if(SHUTTLE_MANIPULATOR_STATUS)
+				dat += "<div class='block'>"
+				for(var/list/shuttle in shuttle_data["shuttles"])
+					dat += "<span class='line'>"
+					dat += "<span class='statusLabel'>[shuttle["name"]] [shuttle["id"]]: [shuttle["mode"]] [shuttle["status"]] [shuttle["timer"]] </span>"
+					dat += "<a href='?src=\ref[src];jump_to=[shuttle["id"]]'>Jump To</a>"
+					dat += "<a href='?src=\ref[src];fast_travel=[shuttle["id"]]'>Fast Travel</a>"
+					dat += "<br></span>"
+				dat += "</div>"
+			if(SHUTTLE_MANIPULATOR_TEMPLATE)
+				dat += "<div class='block'>"
+				for(var/list/shuttle in shuttle_data["shuttles"])
+					dat += "<span class='line'>"
+					dat += "<span class='statusLabel'> [shuttle["name"]] <br>[shuttle["description"]]<br> [shuttle["admin_notes"]] <br></span>"
+					dat += "<A href='?src=\ref[src];load=[shuttle["id"]]'>Load Template</A>"
+					dat += "<A href='?src=\ref[src];select_template=[shuttle["id"]]'>Select Template</A>"
+					dat += "<a href='?src=\ref[src];preview=[shuttle["id"]]'>Preview Template</A>"
+					dat += "<br></span>"
+				dat += "</div>"
+			if(SHUTTLE_MANIPULATOR_MOD)
+				dat += "<div class='block'>"
+				for(var/list/shuttle in shuttle_data["shuttles"])
+					dat += "<span class='line'>"
+					dat += "<span class='statusLabel'>[shuttle["name"]] [shuttle["id"]] <br>[shuttle["description"]]<br> [shuttle["admin_notes"]] <br> </span>"
+					dat += "<A href='?src=\ref[src];move=[shuttle["id"]]'>Send to port</A>"
+					dat += "<a href='?src=\ref[src];jump_to=[shuttle["id"]]'>Jump To</A>"
+					dat += "<br></span>"
+				dat += "</div>"
+
+	var/datum/browser/popup = new(user, "shuttle manipulator", "Shuttle Manipulator", 500, 500)
+	popup.set_content(dat)
+	popup.open()
+
+	return
+
+/obj/machinery/shuttle_manipulator/Topic(href, href_list)
+	if(..())
+		return
+
+	var/mob/user = usr
+
+	// Preload some common parameters
+	var/shuttle_id = href_list["shuttle_id"]
+	var/datum/map_template/shuttle/S = shuttle_templates[shuttle_id]
+
+	if(href_list["select_template"])
+		if(S)
```

Comment:
Oh! Whoops, I overlooked that, but that should work OK, yeah... lemme look at your nano templates, then